acceptance_modelisation/base_acceptance_map_creator.py

```diff
@@ -456,62 +460,67 @@ def _create_model_cos_zenith_binned(self,
+                                                                 wobble_observations == wobble])[0])
+        # Initiate the list of index of selected zenith bin edges
+        zenith_selected = [0]
 
-        # Adapt binning to have a given number of minimum livetime or run in each bin
         i = 0
-        at_least_2_wobble, cut_variable_min_for_each_wobble, condition_is_fulfilled_per_wobble = False, False, False
-
-        while i < len(cut_variable_per_bin):
-            is_last_bin = (i + 1) == len(cut_variable_per_bin)
-            in_coszd_bin = (cos_zenith_observations >= cos_zenith_bin[i]) & (
-                    cos_zenith_observations < cos_zenith_bin[i + 1])
-
-            if per_wobble:
-                wobble_in_bin = wobble_observations[in_coszd_bin]
-                cut_variable_in_bin = cut_variable_weights[in_coszd_bin]
-                cut_variable_in_bin_per_wobble = np.array(
-                    [cut_variable_in_bin[wobble_in_bin == wobble].sum() for wobble in np.unique(wobble_in_bin)])
-                at_least_2_wobble = len(np.unique(wobble_in_bin)) > 1
-                if at_least_2_wobble:
-                    cut_variable_min_for_each_wobble = np.all(
-                        cut_variable_in_bin_per_wobble >= min_cut_per_cos_zenith_bin)
-                condition_is_fulfilled_per_wobble = at_least_2_wobble and cut_variable_min_for_each_wobble
-
-            condition_is_fulfilled_in_bin = cut_variable_per_bin[i] >= min_cut_per_cos_zenith_bin
-
-            if ((per_wobble and not condition_is_fulfilled_per_wobble) or (not condition_is_fulfilled_in_bin)):
-                if not is_last_bin:
-                    cut_variable_per_bin[i] += cut_variable_per_bin[i + 1]
-                    cut_variable_per_bin = np.delete(cut_variable_per_bin, i + 1)
-                    cos_zenith_bin = np.delete(cos_zenith_bin, i + 1)
-                
-                elif i > 0:
-                    cut_variable_per_bin[i - 1] += cut_variable_per_bin[i]
-                    cut_variable_per_bin = np.delete(cut_variable_per_bin, i)
-                    cos_zenith_bin = np.delete(cos_zenith_bin, i)
-                    i -= 1
-                else:
-                    i += 1
+        n = len(cos_zenith_bin) - 2
+
+        while i < n:
+            # For each wobble, find the index of the first zenith which fulfills the zd binning criteria if any
+            # Then flatten and sort the array
+            candidate_i = np.sort((np.concatenate([np.ravel(
+                np.argwhere(cum_cut_variable >= self.cos_zenith_binning_parameter_value))[:1]
+                                                   for cum_cut_variable in cumsum_variable.values()])))
+
+            if len(candidate_i) > multiplicity_wob:
+                # If the criteria is fulfilled save the correct index.
+                # The first and only candidate_i in the non-per_wobble case and the second in the per_wobble case.
```

Comment:
If I've well understood the code, this comment need a correction "and the second in the per_wobble case." -> "and the last (largest one) in the per_wobble case."

Reply:
Currently it is the second . see `i = candidate_i[multiplicity_wob]` where `multiplicity_wob` is 1 in the per_wobble case. So if there is more than 2 wobbles, the current version only impose to have the requested livetime/number of runs for 2 of them.

Reply:
Oh, yes because multiplicity_wob is either 0 or 1. Maybe the name of the variable could be improve but I have no better proposition so far.